Submitted By: Alexander E. Patrakov
Date: 2005-10-10
Initial Package Version: 2.15.94.0.2.2
Upstream Status: Backport from 2.16.1
Origin: Alexander E. Patrakov
Description: Fixes compilation by gcc4 (e.g. from Fedora Core 4 hosts)

--- binutils-2.15.94.0.2.2/gas/tc.h	2004-11-22 20:33:31.000000000 +0000
+++ binutils-2.16.1/gas/tc.h	2005-02-17 13:46:00.000000000 +0000
@@ -24,25 +25,6 @@
 
 extern const pseudo_typeS md_pseudo_table[];
 
-/* JF moved this here from as.h under the theory that nobody except MACHINE.c
-   and write.c care about it anyway.  */
-
-struct relax_type
-{
-  /* Forward reach. Signed number. > 0.  */
-  long rlx_forward;
-  /* Backward reach. Signed number. < 0.  */
-  long rlx_backward;
-
-  /* Bytes length of this address.  */
-  unsigned char rlx_length;
-
-  /* Next longer relax-state.  0 means there is no 'next' relax-state.  */
-  relax_substateT rlx_more;
-};
-
-typedef struct relax_type relax_typeS;
-
 extern const int md_reloc_size;	/* Size of a relocation record.  */
 
 char * md_atof (int, char *, int *);
--- binutils-2.15.94.0.2.2/gas/as.h	2004-09-15 19:05:03.000000000 +0000
+++ binutils-2.16.1/gas/as.h	2005-04-13 17:58:40.000000000 +0000
@@ -397,6 +384,22 @@
 /* Enough bits for address, but still an integer type.
    Could be a problem, cross-assembling for 64-bit machines.  */
 typedef addressT relax_addressT;
+
+struct relax_type
+{
+  /* Forward reach. Signed number. > 0.  */
+  offsetT rlx_forward;
+  /* Backward reach. Signed number. < 0.  */
+  offsetT rlx_backward;
+
+  /* Bytes length of this address.  */
+  unsigned char rlx_length;
+
+  /* Next longer relax-state.  0 means there is no 'next' relax-state.  */
+  relax_substateT rlx_more;
+};
+
+typedef struct relax_type relax_typeS;
 
 /* main program "as.c" (command arguments etc).  */
 
